src/module/token/metadata/BatchMetadataERC721.sol

```diff
@@ -168,6 +177,23 @@ contract BatchMetadataERC721 is Module {
         revert BatchMetadataNoMetadataForTokenId();
     }
 
+    /// @notice sets the metadata for a range of tokenIds.
+    function _setMetadata(uint256 _amount, string calldata _baseURI) internal virtual {
+        if (_amount == 0) {
+            revert BatchMetadataZeroAmount();
+        }
+
+        uint256 rangeStart = _batchMetadataStorage().nextTokenIdRangeStart;
+        uint256 rangeEndNonInclusive = rangeStart + _amount;
+
+        _batchMetadataStorage().nextTokenIdRangeStart = rangeEndNonInclusive;
+        _batchMetadataStorage().tokenIdRangeEnd.push(rangeEndNonInclusive);
+        _batchMetadataStorage().baseURIOfTokenIdRange[rangeEndNonInclusive] = _baseURI;
+
+        emit NewMetadataBatch(rangeStart, rangeEndNonInclusive, _baseURI);
+        emit BatchMetadataUpdate(rangeStart, rangeEndNonInclusive - 1);
```

Comment:
Do we need 2 events here?

Reply:
Probably not, Jake has also mentioned this before as well. I can remove it and rename it to something simpler